app/src/main/java/org/simple/clinic/navigation/v2/fragments/BaseScreen.kt

```diff
@@ -77,6 +84,23 @@ abstract class BaseScreen<K : ScreenKey, B : ViewBinding, M : Parcelable, E, F,
   override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
     super.onViewCreated(view, savedInstanceState)
 
+    val startModel = savedInstanceState?.getParcelable<M>(KEY_MODEL)
+
+    _viewModel = ViewModelProvider(viewModelStore, object : ViewModelProvider.Factory {
+      private fun loop(viewEffectsConsumer: Consumer<V>) = RxMobius
+          .loop(createUpdate(), createEffectHandler())
+          .eventSources(additionalEventSources())
+
+      @Suppress("UNCHECKED_CAST")
+      override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+        return MobiusLoopViewModel.create<M, E, F, V>(
+            ::loop,
+            startModel ?: defaultModel(),
```

Comment:
Can we move the `?: defaultModel()` declaration to where we are reading the model from the saved state. This would be a more readable approach since we are defining the model to start the loop in one statement.